tools/Google.Cloud.Tools.Analyzers/InternalOptionalParametersRequiredAnalyzer.cs

```diff
@@ -106,5 +85,38 @@ private static void AnalyzeInvocation(OperationAnalysisContext context)
                 }
             }
         }
+
+        internal static List<ISymbol> GetVariablesInScope(SyntaxNode syntaxNode, SemanticModel semanticModel) =>
```

Comment:
It may be worth removing the ToList call here, so you only materialize the query where you need to. For example, you don't need to in TryGetVariableForArgument. (You may not need to in the previous code either; I haven't checked.)

I know Roslyn analyzers are probably more performance sensitive than other code - I wouldn't normally mention this...